### ISSUE TITLE
reportUnneccessaryIsInstance reported when doing an exhaustive check

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#3065]
**Describe the bug**
I would like to be to be able to do an exhaustive `isinstance` check without a warning. However, it doesn't seem possible.

I could turn off that warning, but I think it's useful elsewhere.

With `assert_never` being added to `typing` in 3.11, I suspect this pattern may become more common.

**To Reproduce**
```python
def assert_never(x: NoReturn) -> NoReturn:
    assert False, "Unhandled type: {}".format(type(x).__name__)


def what_are_you(value: Union[str, int]):
    if isinstance(value, str):
        print("I'm a str!")
    elif isinstance(value, int):  # Unnecessary isinstance call; "int" is always an instance of "int" Pylance(reportUnnecessaryIsInstance)
        print("I'm an int!")
    else:
        assert_never(value)
```

**Expected behavior**
There to be no warning in this code

**VS Code extension or command-line**
From VS Code, v1.1.222

**Comment by erictraut** [comment]
I'm not sure what you're suggesting here. The check is designed to emit a diagnostic in this circumstance. 

I think your options are:
1. Disable the `reportUnnecessaryIsInstance` diagnostic (throughout your project, on a per-file basis, or on a per-line basis)
2. Don't use this approach for exhaustive checks. Skip the last `elif` clause.
3. Switch to a Python 3.10 `match` statement.

For Python 3.10+ users, I think option 3 is the approach I'd recommend.

**Comment by erictraut** [comment]
I'm really reluctant to add special cases to diagnostic checks like the one you're suggesting. Such special cases make the tool less predictable and more complex.

I'm going to close this for now. I'm willing to revisit this decision if enough other pyright users have a similar request.
